src/libstore/profiles.cc

```diff
@@ -150,27 +158,36 @@ void deleteGenerations(const Path & profile, const std::set<GenerationNumber> &
     }
 }
 
+/**
+ * Advanced the iterator until the given predicate `cond` returns `true`.
+ */
+static inline void iterDropUntil(Generations & gens, auto && i, auto && cond)
+{
+    for (; i != gens.rend() && !cond(*i); ++i);
+}
+
 void deleteGenerationsGreaterThan(const Path & profile, GenerationNumber max, bool dryRun)
 {
+    if (max == 0)
+        throw Error("Must keep at least one generation so we don't delete the current one");
```

Comment:
```suggestion
        throw Error("Must keep at least one generation, otherwise the current one would be deleted");
```

I don't think Nix should say "we" to the user. It's not a research paper (any more).